### ISSUE TITLE
"google.protobuf.Empty" is not defined.

### ISSUE THREAD

**Issue by sivabudh** [grpc/grpc#15937]
### What version of gRPC and what language are you using?
v1.10.1 and C++
 
 
### What operating system (Linux, Windows, …) and version?
 macOS 10.13.5
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
```
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 ```
 
### What did you do?
I created a gRPC `hellostream.proto` file exactly like this:

```
syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

package hellostream;

message UpdateInterval {
  int32 seconds = 1;
}

message TemperatureUnit {
  enum UnitType {
    CELCIUS = 0;
    FARENHEIT = 1;
  }
  UnitType unit = 1;
}

message Temperature {
  google.protobuf.Timestamp timestamp = 1;
  float temperature = 2;
  TemperatureUnit.UnitType unit = 3;
}

service Thermostat {
  rpc GetCurrentTemperature(UpdateInterval) returns (stream Temperature) {}
  rpc ChangeUnitTo(TemperatureUnit) returns (google.protobuf.Empty) {}
}
```
and then I ran the following commands to generate the corresponding `hellostream.grpc.pb.cc/h` and `hellostream.pb.cc/h` files:

```
protoc -I . --grpc_out=. --plugin=protoc-gen-grpc=./vcpkg/bin/grpc_cpp_plugin hellostream.proto
protoc -I . --cpp_out=. hellostream.proto
```
Then, I integrated these into my project and started writing out my service.
 
### What did you expect to see?
That I'm able to build my project fine. 
 
### What did you see instead?
The compiler complained about not being able to find `google.protobuf.Empty` like this:

```
error: no type named 'Empty' in namespace 'google::protobuf'
``` 
### Anything else we should know about your project / environment?

Yes, so after some figuring out, I had to change `hellostream.proto` like this as a workaround and it worked:

```
syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

package hellostream;

message UpdateInterval {
  int32 seconds = 1;
}

message TemperatureUnit {
  enum UnitType {
    CELCIUS = 0;
    FARENHEIT = 1;
  }
  UnitType unit = 1;
}

message Temperature {
  google.protobuf.Timestamp timestamp = 1;
  float temperature = 2;
  TemperatureUnit.UnitType unit = 3;
  google.protobuf.Empty unused = 4;                       // <----------- workaround
}

service Thermostat {
  rpc GetCurrentTemperature(UpdateInterval) returns (stream Temperature) {}
  rpc ChangeUnitTo(TemperatureUnit) returns (google.protobuf.Empty) {}
}
```

Basically, I believe the bug is, if one defines a gRPC service with `google.protobuf.Empty` but none of the `message` included `google.protobuf.Empty` as one of the fields, `protoc` will "forget" to:

```
#include <google/protobuf/empty.pb.h>
```
in `hellostream.pb.h` file.

**Comment by nicolasnoble** [comment]
That's definitely a bug, but a protobuf bug, not a grpc one.

**Comment by nicolasnoble** [comment]
Well, if your analysis is correct that is. But which file exactly failed to compile with this error, I wonder?

**Comment by vjpai** [comment]
Let me add that we highly recommend against using google.protobuf.Empty as
far as best practices go. It is better just to define your own message that
has no fields. The reason is that if you ever decide that you're RPC needs
a request or response type, you can't change it without breaking backwards
compatibility if you use google.protobuf.Empty . In general, it makes the
most sense for each RPC to have its own request and response type defined
even if the current implementation just contains one field of a different
type.

**Comment by sreecha** [comment]
Closing this issue since this is not a grpc bug (but looks like a bug in protobuf)
